api-reference/10 UI Components/dxDataGrid/1 Configuration/summary/calculateCustomSummary.md

```diff
@@ -123,6 +121,202 @@ A summary value calculation is conducted in three stages: *start* - the **totalV
 
 ---
 
+**calculateCustomSummary** is called before [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue). If you want to create a new column and use its values in a custom summary, the following example demonstrates how to implement these functions:
+
+---
+##### jQuery
+
+    <!--JavaScript-->
+    const  calculateArea = (rowData) => {
+        return rowData.width * rowData.height;
+    }
+
+    function calculateAreaSummary(options) {
+        if (options.name === "calculateAreaSummary") {
+            if (options.summaryProcess === "start") {
+                options.totalValue = 0;
+            }
+            if (options.summaryProcess === "calculate") {
+                    options.totalValue += calculateArea(options.value);
+            }
+        }
+    }
+
+    $(function(){    
+        $("#gridContainer").dxDataGrid({
+            // ...
+            columns: [
+                "width", "height",
+                {
+                    dataField: "Area",
+                    calculateCellValue: calculateArea
+                }
+            ],
+            summary: {
+                totalItems: [{
+                        name: "AreaSummary",
+                        summaryType: "custom"
+                        showInColumn: "Area",
+                        displayFormat: "Total Area: {0}",
+                    }
+                ],
+                calculateCustomSummary: calculateAreaSummary
+            }
+        });
+    });
+
+##### Angular
+
+    <!--TypeScript-->
```

Comment:
Use an updated template for code samples, the one where code is split across tabs named like files in a real project.